examples/c-guest/Makefile

```diff
@@ -0,0 +1,53 @@
+ROOT_DIR:=$(strip $(dir $(realpath $(lastword $(MAKEFILE_LIST)))))
+
+ifeq ($(OS),Windows_NT)
+$(error Windows not supported)
+else
+    UNAME_S := $(shell uname -s)
+    ifeq ($(UNAME_S),Linux)
+        MACHINE = linux
+    endif
+    ifeq ($(UNAME_S),Darwin)
+        MACHINE = osx
+    endif
+
+    UNAME_P := $(shell uname -p)
+    ifeq ($(UNAME_P),x86_64)
+        MACHINE := $(MACHINE)-x86_64
+    endif
+    ifneq ($(filter arm%,$(UNAME_P)),)
+        MACHINE := $(MACHINE)-arm64
+    endif
+endif
+
+default: execute
+
+.PHONY: platform
+platform:
+	cargo +risc0 rustc -p zkvm-platform --target riscv32im-risc0-zkvm-elf --lib --crate-type staticlib --release --target-dir ./guest/out/platform
+
+# Regenerate the header file. Assumes cbindgen installed locally.
+.PHONY: headers
+headers:
+    # Note: this is currently broken on main, as there is a FFI incompatible type
```

Comment:
Noticed on main that this no longer works (something in this crate is not FFI compatible since 1.0.0-rc.5.

Not a big deal and this obviously isn't a constraint that previously mattered, just pointing out :)